src/segment-loader.js

```diff
@@ -741,31 +740,58 @@ export default class SegmentLoader extends videojs.EventTarget {
       return false;
     }
 
+    const currentTime = this.currentTime_();
     const measuredBandwidth = stats.bandwidth;
+    const segmentDuration = this.pendingSegment_.duration;
     const estimatedSegmentSize =
-      this.pendingSegment_.duration * this.playlist_.attributes.BANDWIDTH;
+      segmentDuration * this.playlist_.attributes.BANDWIDTH;
     const requestTimeRemaining =
       (estimatedSegmentSize - (stats.bytesReceived * 8)) / measuredBandwidth;
     const buffered = this.buffered_();
     const bufferedEnd = buffered.length ? buffered.end(buffered.length - 1) : 0;
     // Subtract 1 from the timeUntilRebuffer so we still consider an early abort
     // if we are only left with less than 1 second when the request completes.
     const timeUntilRebuffer =
-      ((bufferedEnd - this.currentTime_()) / this.hls_.tech_.playbackRate()) - 1;
+      ((bufferedEnd - currentTime) / this.hls_.tech_.playbackRate()) - 1;
 
     // Only consider aborting early if the estimated time to finish the download
     // is larger than the estimated time until the player runs out of forward buffer
     if (requestTimeRemaining <= timeUntilRebuffer) {
       return false;
     }
 
-    const { playlist, roundTripTime } =
-      minRebufferingSelector(this.hls_.playlists.master,
-                             measuredBandwidth,
-                             timeUntilRebuffer,
-                             this.pendingSegment_.duration);
+    const switchCandidate = this.hls_.playlists.master.playlists.filter(
+      (playlist) => playlist.attributes && playlist.attributes.BANDWIDTH
+    ).map((playlist) => {
+      const syncPoint = this.syncController_.getSyncPoint(playlist,
+                                                          this.duration_(),
+                                                          this.currentTimeline_,
+                                                          currentTime);
+      // There is no sync point for this playlist, so switching to it will require a
+      // sync request first. This will double the round trip time
+      const numRequests = syncPoint ? 1 : 2;
+      const segmentSize = segmentDuration * playlist.attributes.BANDWIDTH;
```

Comment:
Minor, but the segments on different playlists might have different durations, thus could have a small impact. I doubt it will be noticeable, but it's possible. (Same for two request sets, but probably isn't worth worrying too much about.)